### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by reubenharry**
This PR does a bit of cleaning:

- I make imports explicit
- I made a few language extensions global defaults (see the .cabal)
- I remove a few functions that aren't being used in `Weighted` and `Population`

### CODE REVIEW

**Comment by reubenharry** [comment]
@idontgetoutmuch  This is ready to be reviewed and merged, after the docs PR.

**Comment by reubenharry** [comment]
@idontgetoutmuch OK, I merged in main, so this is ready to review

**Comment by reubenharry** [comment]
Actually, something weird happened when I ran the benchmarks, so let me iron that out first. I've made this a draft for the time being.

**Comment by reubenharry** [comment]
Confirming that this is ready to review

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,23 @@
 
 - No unreleased changes so far.
 
+# 0.1.1.2 (2021-06-08)
+
+Clean up of unused functions and broken code
+
+- remove unused functions in `Weighted` and `Population`
```
Comment (idontgetoutmuch) [inline CHANGELOG.md:9]: Do we know what these did? I guess we should raise a ticket?

Comment (idontgetoutmuch) [reply CHANGELOG.md:9]: https://github.com/tweag/monad-bayes/issues/146

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,23 @@
...
+- remove broken models in `models`
```
Comment (idontgetoutmuch) [inline CHANGELOG.md:10]: I will add tickets for these when I reach them.

File: docs/build/html/genindex.html [hunk 3]
```diff
@@ -165,7 +165,7 @@ <h1 id="index">Index</h1>
 
   <div role="contentinfo">
     <p>
-        &copy; Copyright 2021, Adam Scibior.
+        &copy; Copyright 1980, Adam Scibior.
```
Comment (idontgetoutmuch) [inline docs/build/html/genindex.html:168]: ?

File: docs/build/html/index.html [hunk 4]
```diff
@@ -198,7 +198,7 @@ <h1>Welcome to monad-bayes’s documentation!<a class="headerlink" href="#welcom
 
   <div role="contentinfo">
     <p>
-        &copy; Copyright 2021, Adam Scibior.
+        &copy; Copyright 1980, Adam Scibior.
```
Comment (idontgetoutmuch) [inline docs/build/html/index.html:201]: ?

File: docs/build/html/probprog.html [hunk 5]
```diff
@@ -599,7 +599,7 @@ <h2>API docs<a class="headerlink" href="#api-docs" title="Permalink to this head
 
   <div role="contentinfo">
     <p>
-        &copy; Copyright 2021, Adam Scibior.
+        &copy; Copyright 1980, Adam Scibior.
```
Comment (idontgetoutmuch) [inline docs/build/html/probprog.html:602]: ?

File: docs/build/html/search.html [hunk 6]
```diff
@@ -169,7 +169,7 @@
 
   <div role="contentinfo">
     <p>
-        &copy; Copyright 2021, Adam Scibior.
+        &copy; Copyright 1980, Adam Scibior.
```
Comment (idontgetoutmuch) [inline docs/build/html/search.html:172]: ?

File: docs/build/html/usage.html [hunk 7]
```diff
@@ -849,7 +849,7 @@ <h3>Sequential Monte Carlo squared ($SMC^2$)<a class="headerlink" href="#sequent
 
   <div role="contentinfo">
     <p>
-        &copy; Copyright 2021, Adam Scibior.
+        &copy; Copyright 1980, Adam Scibior.
```
Comment (idontgetoutmuch) [inline docs/build/html/usage.html:852]: ?

File: monad-bayes.cabal [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
-version:            0.1.1.0
+version:            0.1.1.2
 license:            MIT
 license-file:       LICENSE.md
 copyright:          2015-2020 Adam Scibior
 maintainer:         dominic.steinitz@tweag.io
 author:             Adam Scibior <adscib@gmail.com>
 stability:          experimental
-tested-with:        GHC ==8.4.4 || ==8.6.5 || ==8.8.1
+tested-with:        GHC ==9.2.2
```
Comment (idontgetoutmuch) [inline monad-bayes.cabal:10]: We probably ought to do something about this: we should test for 3 versions of ghc I think.

File: CHANGELOG.md (hunk 2 above)
Comment (idontgetoutmuch) [reply CHANGELOG.md:10]: I didn't find these: maybe you (@reubenharry) can raise a ticket?

File: monad-bayes.cabal (hunk 8 above)
Comment (idontgetoutmuch) [reply monad-bayes.cabal:10]: https://github.com/tweag/monad-bayes/issues/147

File: CHANGELOG.md (hunk 2 above)
Comment (reubenharry) [reply CHANGELOG.md:10]: Didn't find what? The removed functions?

File: docs/build/html/genindex.html (hunk 3 above)
Comment (reubenharry) [reply docs/build/html/genindex.html:168]: Weird. How on earth did that happen..

**Review by idontgetoutmuch — APPROVED** [review]

**Pull request merged**